sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/geojson/GeoArrayIterator.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.geojson;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+final class GeoArrayIterator<T> implements Iterator<T> {
+    private final ClientLogger logger = new ClientLogger(GeoArrayIterator.class);
+
+    private final GeoArray<T> container;
+
+    private transient int cursor;
```

Comment:
Why transient? Do we expect this iterator to be serialized?

Reply:
I believe I meant to make this volatile, good catch